pkgs/test/haxe/default.nix

```diff
@@ -0,0 +1,175 @@
+        nativeBuildInputs = [ haxePackages.haxe haxePackages.hxcpp ];
+        buildPhase = "haxe build-cpp.hxml";
+        installPhase = "install -Dt ${placeholder "out"} dist/cpp/Main";
+        doInstallCheck = true;
+        installCheckPhase = "${placeholder "out"}/Main";
+      })
+    {
+      cpp        = { stdenv = stdenv; };
+      cpp-32     = { stdenv = pkgsi686Linux.stdenv; };
+      cpp-static = { stdenv = pkgsStatic.stdenv; };
+      cpp-musl   = { stdenv = pkgsMusl.stdenv; };
+      cpp-clang  = { stdenv = llvmPackages.stdenv; };
+    }
+  ) // (
+    lib.mapAttrs (name: {php}:
+      stdenv.mkDerivation {
+        pname = "haxe-example-${name}";
+        inherit (haxePackages.haxe) version;
+        inherit src;
+        nativeBuildInputs = [ haxePackages.haxe ];
+        buildPhase = "haxe build-php.hxml";
+        installPhase = "cp -r dist/php ${placeholder "out"}";
+        doInstallCheck = true;
+        installCheckInputs = [ php ];
+        installCheckPhase = "php ${placeholder "out"}/index.php";
+      })
+    {
+      php74 = { php = php74; };
+      php80 = { php = php80; };
+    }
+  ) // (
+    lib.mapAttrs (name: {luaPackages}:
+      stdenv.mkDerivation {
+        pname = "haxe-example-${name}";
+        inherit (haxePackages.haxe) version;
+        inherit src;
+        nativeBuildInputs = [ haxePackages.haxe ];
+        buildPhase = "haxe build-lua.hxml";
+        installPhase = "install -Dm444 -t ${placeholder "out"} dist/build.lua";
+        doInstallCheck = true;
+        installCheckInputs = [ luaPackages.lua luaPackages.luautf8 ];
+        installCheckPhase = "lua ${placeholder "out"}/build.lua";
+      })
+    {
+      lua52  = { luaPackages = lua52Packages;  };
+      lua53  = { luaPackages = lua53Packages;  };
+      luajit = { luaPackages = luajitPackages; };
+    }
+  ) // (
+    lib.mapAttrs (name: {jre}:
+      stdenv.mkDerivation {
+        pname = "haxe-example-${name}";
+        inherit (haxePackages.haxe) version;
+        inherit src;
+        nativeBuildInputs = [ haxePackages.haxe haxePackages.hxjava ];
+        buildPhase = "haxe build-java.hxml";
+        installPhase = "install -Dm444 -t ${placeholder "out"} dist/java/Main.jar";
+        doInstallCheck = true;
+        installCheckInputs = [ jre ];
+        installCheckPhase = "java -jar ${placeholder "out"}/Main.jar";
+      })
+    {
+      java8  = { jre = jre8;  };
+      java11 = { jre = jdk11; };
+    }
+  ) // (
+    lib.mapAttrs (name: {mono}:
+      stdenv.mkDerivation {
+        pname = "haxe-example-${name}";
+        inherit (haxePackages.haxe) version;
+        inherit src;
+        nativeBuildInputs = [ haxePackages.haxe haxePackages.hxcs ];
+        buildPhase = "haxe build-cs.hxml";
+        installPhase = "install -Dm444 -t ${placeholder "out"} dist/cs/bin/Main.exe";
+        doInstallCheck = true;
+        installCheckInputs = [ mono ];
+        installCheckPhase = "mono ${placeholder "out"}/Main.exe";
```

Comment:
```suggestion
        installPhase = "install -Dm444 -t $out dist/cs/bin/Main.exe";
        doInstallCheck = true;
        installCheckInputs = [ mono ];
        installCheckPhase = "mono $out/Main.exe";
```

Reply:
Is there any discussion about `${placeholder "out"}` vs environment `$out` ?
I notice slow migration towards the former in nixpkgs. Not only in new code, but also changes from environment to placeholder in existing code